liquibase-core/src/main/java/liquibase/command/AbstractCommandStep.java

```diff
@@ -1,29 +1,23 @@
 package liquibase.command;
 
 import liquibase.exception.CommandValidationException;
-import liquibase.util.StringUtil;
 
-import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * Convenience base class for {@link CommandStep} implementations.
  */
 public abstract class AbstractCommandStep implements CommandStep {
 
-    /**
-     * @return {@link #ORDER_DEFAULT} if the command scope's name matches {@link #defineCommandNames()}. Otherwise {@link #ORDER_NOT_APPLICABLE}
-     */
     @Override
-    public int getOrder(CommandDefinition commandDefinition) {
-        final String[][] definedCommandNames = defineCommandNames();
-        if (definedCommandNames != null) {
-            for (String[] thisCommandName : definedCommandNames) {
-                if ((thisCommandName != null) && StringUtil.join(Arrays.asList(thisCommandName), " ").equalsIgnoreCase(StringUtil.join(Arrays.asList(commandDefinition.getName()), " "))) {
-                    return ORDER_DEFAULT;
-                }
-            }
-        }
-        return ORDER_NOT_APPLICABLE;
+    public List<Class<?>> requiredDependencies() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public List<Class<?>> providedDependencies() {
+        return Collections.emptyList();
```

Comment:
Do we have a plan in mind for how to gracefully migrate any extensions or users who have classes which extend this class and make use of the `getOrder` method?

Reply:
Does the pro code compile successfully with the removal of the getOrder method?

Reply:
I think we'll need to deprecate `getOrder`for a while and not remove it yet. I wasn't able to build core and pro with this change, and I'd expect some other extensions may throw a fit too.

Reply:
I only verified core code and it does not use order, but I forgot that pro does. I rolledback and deprecated getOrder, also changing the logic to work with order and dependencies. I created a branch for this issue on pro and it is building now.
I think that in the next 2 or 3 commands this solution will become stable and then we can go ahead and refactor pro/other extensions code to get rid of getOrder.